app-backend/akkautil/src/main/scala/Authentication.scala

```diff
@@ -173,14 +174,30 @@ trait Authentication extends Directives with LazyLogging {
             platform: Option[Platform],
             user: User
         )
-        // All users will have a platform role, either added by a migration or created with the user if they are new
-        val query = for {
-          (user, roles) <- OptionT[ConnectionIO, (User, List[UserGroupRole])] {
-            UserDao.getUserAndActiveRolesById(userId).flatMap {
+
+        val getOrCreateUserProgram = for {
+          u <- OptionT.liftF(
+            UserDao.getUserAndActiveRolesById(userId).transact(xa)
+          )
+          (user, roles) <- OptionT {
+            u match {
               case UserOptionAndRoles(Some(user), roles) =>
-                Option((user, roles)).pure[ConnectionIO]
+                IO.pure(Some((user, roles)))
               case UserOptionAndRoles(None, _) => {
                 createUserWithRoles(userId, email, name, picture, jwtClaims)
+                  .transact(xa)
+                  .attempt flatMap {
+                  case Right(s) => IO.pure(s)
+                  case Left(e: PSQLException) if e.getSQLState == "23505" => {
+                    logger.debug(s"Handling Duplicate User: $e")
```

Comment:
The way this ends up printed is `Handling Duplicate User: {}` -- I think we can probably throw away the error itself in the log if that's how it ends up printing
```suggestion
                    logger.debug("Handling Duplicate User")
```